src/event-handlers/impl/ProjectContentEventHandler.js

```diff
@@ -75,28 +75,41 @@ projectContentEventHandler.register(APP_EVENT.PROJECT_CONTENT_DRAFT_CREATED, asy
     foreignReferences: []
   }
 
-  const projectContentPackageDirPath = FileStorage.getResearchContentPackageDirPath(projectId, externalId);
-  const hashObj = await FileStorage.calculateDirHash(projectContentPackageDirPath, options);
-  const hashes = hashObj.children.map(f => f.hash);
-  hashes.sort();
-  const packageHash = crypto.createHash('sha256').update(hashes.join(",")).digest("hex");
-  draftData.hash = packageHash;
-  draftData.packageFiles = hashObj.children.map((f) => ({ filename: f.name, hash: f.hash, ext: path.extname(f.name) }));
+  if (formatType === PROJECT_CONTENT_DATA_TYPES.JSON) {
+    const packageHash = crypto.createHash('sha256').update(JSON.stringify(jsonData)).digest("hex");
+    draftData.jsonData = jsonData;
+    draftData.hash = packageHash;
+    draftData.packageFiles = [];
+  } else {
+    const projectContentPackageDirPath = FileStorage.getResearchContentPackageDirPath(projectId, externalId);
+    const hashObj = await FileStorage.calculateDirHash(projectContentPackageDirPath, options);
+    const hashes = hashObj.children.map(f => f.hash);
+    hashes.sort();
+    const packageHash = crypto.createHash('sha256').update(hashes.join(",")).digest("hex");
+    draftData.hash = packageHash;
```

Comment:
Let's use genSha256Hash from the @deip/toolbox package everywhere we need the hash